a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -0,0 +1,201 @@
+    'The number of active vaults should remain the same.',
+  );
+});
+
+test.serial('open new vault', async t => {
+  const istBalanceBefore = await getISTBalance(USER1ADDR);
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '5.0';
+  const collateral = '10.0';
+  await openVault(USER1ADDR, mint, collateral);
+  await waitForBlock();
+
+  const istBalanceAfter = await getISTBalance(USER1ADDR);
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+
+  t.true(
+    istBalanceBefore + 5 === istBalanceAfter,
+    'The IST balance should increase by the minted amount',
+  );
+  t.true(
+    activeVaultsAfter.length > activeVaultsBefore.length,
+    `The number of active vaults should increase after opening a new vault.`,
+  );
+});
+
+test.serial('remove collateral', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultDataString = await queryVstorage(vaultPath);
+  let vaultData = getFormattedVaultData(vaultDataString);
+  const collateralBefore = vaultData.locked.value;
+
+  await adjustVault(USER1ADDR, vaultID, { wantCollateral: 1.0 });
+  await waitForBlock();
+
+  vaultDataString = await queryVstorage(vaultPath);
+  vaultData = getFormattedVaultData(vaultDataString);
+  const collateralAfter = vaultData.locked.value;
+
+  t.true(
+    Number(collateralBefore) > Number(collateralAfter),
+    'The vault Collateral should decrease after removing some ATOM',
+  );
+});
+
+test.serial('remove IST', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultDataString = await queryVstorage(vaultPath);
+  let vaultData = getFormattedVaultData(vaultDataString);
+  const debtBefore = vaultData.debtSnapshot.debt.value;
+
+  await adjustVault(USER1ADDR, vaultID, { wantMinted: 1.0 });
+  await waitForBlock();
+
+  vaultDataString = await queryVstorage(vaultPath);
+  vaultData = getFormattedVaultData(vaultDataString);
+  const debtAfter = vaultData.debtSnapshot.debt.value;
+
+  t.true(
+    Number(debtAfter) > Number(debtBefore),
+    'The vault Debt should increase after removing some IST',
+  );
+});
+
+test.serial('close vault', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  await closeVault(USER1ADDR, vaultID, 6.03);
+  await waitForBlock();
+
+  const vaultDataString = await queryVstorage(vaultPath);
+  const vaultData = getFormattedVaultData(vaultDataString);
+  const vaultState = vaultData.vaultState;
+
+  t.is(vaultState, 'closed', 'The vault should be in the "closed" state.');
```

Comment:
Please assert that assets were returned.

Reply:
Thank you for pointing this out.

At this commit, I created an `utils.js` file that expose the `getBalance` function that is also imported by the `wallet.test.js`.
As well as other functions that are used by both tests.

Change addressed on commit [298654f](https://github.com/Agoric/agoric-sdk/pull/10123/commits/298654fda24641bf4310121c2cbf163967b54933)